mnefun/_mnefun.py

```diff
@@ -2122,4 +2122,22 @@ def _headpos(p, file_in, file_out):
     if not op.isfile(file_out):
         run_sss_positions(file_in, file_out, host=p.sws_ssh)
     pos = read_head_pos(file_out)
-    return pos
\ No newline at end of file
+    return pos
+
+
+def _get_ave_trans(pos, mode='mean'):
+    """Helper for computing average head transformation matrix
+    without rotation from cHPI data"""
+    assert pos.shape[1] == 10
+    t = pos[:, 0]
+    # q4:q6 cols from Maxfilter hp file
+    trans = pos[:, 4:7]
+    # time weighting factor as col vector
+    frac = np.diff(t)[:, np.newaxis] / (t[-1] - t[0])
+    assert frac.size == trans[:-1].shape[0]
+    if mode == 'mean':
```

Comment:
For future if & when we want an option between mean or median hp.
On Mar 15, 2016 12:11 PM, "Eric Larson" <notifications@github.com> wrote:

> In mnefun/_mnefun.py
> <https://github.com/LABSN/mnefun/pull/118#discussion_r56223087>:
>
> > \ No newline at end of file
> > +    return pos
> > +
> > +
> > +def _get_ave_trans(pos, mode='mean'):
> > +    """Helper for computing average head transformation matrix
> > +    without rotation from cHPI data"""
> > +    assert pos.shape[1] == 10
> > +    t = pos[:, 0]
> > +    # q4:q6 cols from Maxfilter hp file
> > +    trans = pos[:, 4:7]
> > +    # time weighting factor as col vector
> > +    frac = np.diff(t)[:, np.newaxis] / (t[-1] - t[0])
> > +    assert frac.size == trans[:-1].shape[0]
> > +    if mode == 'mean':
>
> why the conditional? if mode != 'mean' then p0 won't even exist...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/LABSN/mnefun/pull/118/files/e7dc66438466aaa7ba7e1730e56c0e05baa6fb9d#r56223087>
>